STYLE.md

```diff
@@ -0,0 +1,39 @@
+# dqlite style guide
+
+## Introduction
+
+This document records style guidance for new dqlite code. Due to inconsistent
+style preferences by the authors over time and imperfect enforcement, existing
+code may not follow these guidelines.
+
+## Enforcement
+
+The repository includes `.clang-format` and `.clang-tidy` files for use with
+the corresponding LLVM tools. These can be used to enforce many of the
+guidelines listed below. The remaining guidelines are intended to be enforced
+manually by authors and maintainers, or broken when necessary for practical or
+aesthetic reasons.
+
+## Guidelines
```

Comment:
1. Agreed.
2. We use explicit comparisons to `NULL` and 0 in the current code, so let's stick with that.
3. Agreed, obviously there's a lot of macro stuff in the current codebase but let's try to limit that going forward.
4. This is tough because basically all our functions use positive return codes to signal errors right now. We definitely need some kind of reform here because (similar to the allocator situation) our functions are split between using raft return codes, SQLite codes, dqlite codes, and I think libuv codes in a few cases. But I'm not sure what our stated policy should be in this document.
5. Agreed.
6. Sure!